modules/core/include/opencv2/core/types.hpp

```diff
@@ -57,6 +57,12 @@
 #include "opencv2/core/cvstd.hpp"
 #include "opencv2/core/matx.hpp"
 
+#if defined(_MSC_VER) && (_MSC_VER <= 1800)
+#define CV_NOEXCEPT
+#else
+#define CV_NOEXCEPT noexcept
+#endif
```

Comment:
Could you move this into cvdef.h file please?
Similar to [this](https://github.com/opencv/opencv/blob/3.4.2/modules/core/include/opencv2/core/cvdef.h#L440-L453)